NumberToPairValidation.c

```diff
@@ -0,0 +1,45 @@
+#include <stdio.h>
+#include <assert.h>
+
+#include "ColorCoding.h"
+#include "NumberToPairValidation.h"
+
+/**********************************************************************************************
+ * Global variables
+ *********************************************************************************************/
+extern const char* MajorColorNames[];
+extern const char* MinorColorNames[];
+extern int numberOfMinorColors;
+
+//Function to test the number to color pair
+void ValidateNumberToPair(int pairNumber, MajorColor expectedMajor, MinorColor expectedMinor)
+{
+	char colorPairNames[MAX_COLORPAIR_NAME_CHARS];
+
+	ColorPair colorPair = GetColorFromPairNumber(pairNumber);
+    ColorPairToString(&colorPair, colorPairNames);
+    printf("Got pair %s\n", colorPairNames);
+
+    //verify the color with the expected colors
+    assert(colorPair.majorColor == expectedMajor);
+    assert(colorPair.minorColor == expectedMinor);
+}
+
+//Function to derive the Color from the PairNumber
+ColorPair GetColorFromPairNumber(int pairNumber)

```

Comment:
Keep the validation and the functionality in different files. Otherwise re-use of functionality means the test code also gets included